### PR TITLE
interface: convert api to http-api package

### PR DESCRIPTION
**Pull request by tylershuster**
This PR converts the global, prop-drilled API to a pseudo-hooked api that uses the `npm/http-api` package to send pokes, threads, and scrys as defined by the `npm/api` helper functions.

This is still under heavy development but works with a bit of finagling. Notable problems include not getting new messages after sending one in chat...? and initial connection issues. However, you can see the general patterns. Lots going on here, but I think this really all has to go in one go.

It also adds some functionality into `npm/http-api` itself (need to bump version) that I'm still working out to bring to parity with `channel.js`.

For ergonomics' sake, I've also updated the exports in `npm/api` so as to use the following pattern:
```ts
import { settings } from '@urbit/api';
api.poke(settings.putEntry(...));
```

cc @matildepark @tacryt-socryp @liam-fitzgerald

### CODE REVIEW

**Comment by tylershuster** [comment]
Remaining items, but this seems to work generally:

- [ ] Initial hark-update seems to fail? (no notification counts)
- [ ] New messages trigger a spinner in chat
- [ ] Make sure GCP works

**Comment by tylershuster** [comment]
Apparently the order of subscriptions matters

**Comment by tylershuster** [comment]
Ready for review with the exception of the endless stream of merge conflicts that will occur and the issue with pending message resolution in chat

**Comment by tylershuster** [comment]
Ok, there's some strange connection loss but functionally it's all g2g.

**Comment by matildepark** [comment]
So what I’m thinking is pending last hark fixes we’ll stage next-js to livenet and next-userspace early next week and stage this on devnet. 

Though I’m afraid of the merge conflict that may result when next userspace hits master, getting merged into next-js. I guess we deal with that then.

**Comment by tylershuster** [comment]
Is now the time?

**Comment by matildepark** [comment]
I'm aiming for a Friday stage, there's a regression we want to stage overnight for a Thurs release ... sorry.

**Comment by tylershuster** [comment]
Latest commit addresses the new fetch API throwing a "channel already bound" error when leaving and returning to the window where it is being used.

**Comment by matildepark** [comment]
@tylershuster Should we get ready to stage and test this, then?

**Comment by tylershuster** [comment]
Well well well if it isn't the consequences of my own actions

**Review by tacryt-socryp — APPROVED** [review]
Looks beautiful, let's see how it feels on release/next-js

File: pkg/interface/src/logic/reducers/contact-update.ts [hunk 1]
```diff
@@ -5,25 +5,29 @@ import { ContactUpdate } from '@urbit/api';
...
   if (data) {
-    reduceState<ContactState, ContactUpdate>(useContactState, data, [
-      initial,
-      add,
-      remove,
-      edit,
-      setPublic
-    ]);
+    useContactState.getState().set(state => {
+      state = reduceState<ContactState, ContactUpdate>(useContactState, data, [
```
Comment (liam-fitzgerald) [inline pkg/interface/src/logic/reducers/contact-update.ts:16]: !! please don't assign to immer generated variables, this breaks it

File: pkg/interface/src/logic/reducers/gcp-reducer.ts [hunk 2]
```diff
@@ -7,10 +7,12 @@ import { reduceState } from '../state/base';
 
 export default class GcpReducer {
   reduce(json: Cage) {
-    reduceState<StorageState, any>(useStorageState, json, [
-      reduceConfigured,
-      reduceToken
-    ]);
+    useStorageState.getState().set(state => {
+      state = reduceState<StorageState, any>(useStorageState, json, [
```
Comment (liam-fitzgerald) [inline pkg/interface/src/logic/reducers/gcp-reducer.ts:11]: same

File: pkg/interface/src/logic/state/base.ts [hunk 3]
```diff
@@ -24,11 +24,13 @@ export const reduceState = <
   state: UseStore<StateType>,
   data: UpdateType,
   reducers: ((data: UpdateType, state: StateType) => StateType)[]
-): void => {
+): StateType=> {
   const oldState = state.getState();
   const reducer = compose(reducers.map(reducer => reducer.bind(reducer, data)));
   const newState = reducer(oldState);
-  state.getState().set(state => state = newState);
```
Comment (liam-fitzgerald) [inline pkg/interface/src/logic/state/base.ts:31]: again.
consider `Object.assign(state, newState)`

**Review by liam-fitzgerald — CHANGES_REQUESTED** [review]
Please fix assigning to immer top level variables. Other than that this is a good and righteous change.

File: pkg/interface/src/logic/reducers/contact-update.ts (hunk 1 above)
Comment (tylershuster) [reply pkg/interface/src/logic/reducers/contact-update.ts:16]: Yep, these are all marked as merge conflicts anyway. Will use new pattern

**Comment by tylershuster** [comment]
This has gotten pretty bonkers with all the new work that's gone in. I think I should probably just publish the API/HTTP-API changes I've made while learning this and then work on rolling it in in yet another PR.

**Pull request closed without merging**